lmms_eval/models/llava.py

```diff
@@ -198,7 +198,7 @@ def loglikelihood(self, requests: List[Instance]) -> List[Tuple[float, bool]]:
             else:
                 image = None
 
-            prompts_input = contexts[0]
+            prompts_input = contexts[0] if isinstance(contexts, list) else contexts
```

Comment:
OK this didn't have much impact after all: `seedbench_ppl` went from `0.168` -> `0.112`.

What is odd is that the numbers reported in the [paper](https://arxiv.org/pdf/2310.03744.pdf) are much higher ~0.6 which suggests there is also an issue in the original `llava.py` implementation as well:

![Screenshot 2024-04-09 at 16 13 22](https://github.com/EvolvingLMMs-Lab/lmms-eval/assets/26859204/6ca002fb-e15b-4aa5-899d-5a9fe904653e)

For reference, this is the command I am running:

```shell
accelerate launch --num_processes=8 -m lmms_eval --model llava   --model_args pretrained=liuhaotian/llava-v1.5-7b   --tasks seedbench_ppl --batch_size 1 --output_path ./logs/ --log_samples
```

Perhaps this is something that can be dealt with in a follow up PR?

Reply:
Hmmm, I also notice this huge result difference when I write the seedbench ppl. The original llava use generation instead of ppl to generate answer and I achieve similar result using the generation version. So I feel this is just whether you use generation or perplexity. May I ask in your seedbench ppl logs, are the answer being matched correctly?